plugins/experimental/ssl_session_reuse/src/config.cc

```diff
@@ -0,0 +1,149 @@
+/** @file
+
+  config.cc - config file support
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+ */
+
+#include <ts/ts.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include "tscpp/util/TextView.h"
+
+#include "Config.h"
+#include "common.h"
+
+Config::Config()
+{
+  m_filename = "";
+  m_config.clear();
+  m_noConfig      = false;
+  m_alreadyLoaded = false;
+  m_lastCheck     = 0;
+  m_lastmtime     = 0;
+}
+
+Config::~Config() {}
+
+bool
+Config::loadConfig(const std::string &filename)
+{
+  if (m_alreadyLoaded) {
+    return true;
+  }
+
+  bool success = false;
+
+  m_filename = filename;
+
+  int fd = (this->m_filename.length() > 0 ? open(m_filename.c_str(), O_RDONLY) : 0);
+  struct stat info;
+  if (0 == fstat(fd, &info)) {
+    size_t n = info.st_size;
+    std::string config_data;
+    config_data.resize(n);
+    read(fd, const_cast<char *>(config_data.data()), n);
+
+    ts::TextView content(config_data);
+    while (content) {
+      ts::TextView line = content.take_prefix_at('\n');
```

Comment:
I'd add `.ltrim_if(&isspsace)` to clean out leading whitespace. That will make lines containing only white space empty.